### PR TITLE
Add a11y documentation for Icons in storybook

### PR DESCRIPTION
**Pull request by nishasy**
## Summary:
Conduct research and add an accessibility section to the
Storybook documentation for wonder-blocks-icon.

I added a list of rules as well as two relevant demos.

Issue: https://khanacademy.atlassian.net/browse/WB-1361

## Test plan:
- Double check the documentation looks right at http://localhost:63824/?path=/docs/icon-accessibility--right-to-left-icons
- Click on the reference links at the bottom and confirm
  the link to the articles they're labeled as.
- Use a contrast checker to confirm that the blue icon passes
  and the light blue icon fails, as claimed.

image

### CODE REVIEW

**Comment by nishasy** [comment]
If anyone has any other points about icons that they think I should add, please let me know!

File: __docs__/wonder-blocks-icon/accessibility.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+-   Icons should be easy to see.
+-   The color contrast should pass WCAG.
+    -   At least 4.5:1 contrast ratio for icons.
+-   Ideally, an icon should be accompanied by text. However, if an icon
+    is not accompanied by text, provide alt text.
+    -   Do not provide alt text for decorative icons. Set the alt text
+        to an empty string (`""`).
+    -   The `title` attribute is often used to display a description on mouse
+        hover, but please note that use of the `title` attribute is discouraged
+        as titles are not available to keyboard-only or touch-only users.
```
Comment (jandrade) [inline __docs__/wonder-blocks-icon/accessibility.stories.mdx:38]: **thought:** What do you think about mentioning that there's more info in the `References` section?

File: __docs__/wonder-blocks-icon/accessibility.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import {Meta, Story, Canvas} from "@storybook/addon-docs";
+
+import Color from "@khanacademy/wonder-blocks-color";
```
Comment (jandrade) [inline __docs__/wonder-blocks-icon/accessibility.stories.mdx:3]: **suggestion:** It looks like `import {View} from wb-core` is missing here. I was inspecting the HTML and noticed that the `View` styles defined in this file are not being applied correctly, so I'm not sure if this is related to the missing import.

Screen Shot 2023-03-14 at 10 50 50 AM

**Review by jandrade — APPROVED** [review]
Looks great! 🚢 

Thanks for adding all these a11y details, I'm sure folks will find it super helpful! I added a couple of (unrelated) notes regarding the styling of the stories and a minor copy change for you to consider.

File: __docs__/wonder-blocks-icon/accessibility.stories.mdx (hunk 2 above)
Comment (nishasy) [reply __docs__/wonder-blocks-icon/accessibility.stories.mdx:3]: Oh!! Good catch!

File: __docs__/wonder-blocks-icon/accessibility.stories.mdx (hunk 1 above)
Comment (nishasy) [reply __docs__/wonder-blocks-icon/accessibility.stories.mdx:39]: Did you mean this specifically for this bullet point? Or for the entire list?

Comment (jandrade) [reply __docs__/wonder-blocks-icon/accessibility.stories.mdx:39]: yep, specifically to this bullet point.

**Pull request merged**